frontend/src/app/App.tsx

```diff
@@ -67,9 +72,37 @@ const App: React.FC = () => {
     [buildStatuses, dashboardConfig, storageClasses],
   );
 
+  const isUnauthorized = React.useMemo(() => {
+    if (fetchConfigError?.request?.status === 403) {
+      return true;
+    }
+    return false;
+  }, [fetchConfigError]);
+
   // We lack the critical data to startup the app
   if (userError || fetchConfigError) {
-    // There was an error fetching critical data
+    // Check for unauthorized state
+    if (isUnauthorized) {
+      return (
+        <Modal variant={ModalVariant.small} isOpen ouiaId="BasicModal">
+          <ModalHeader title="Session Expired" titleIconVariant="warning" />
+          <ModalBody data-testid="timeout-text">
+            Your session timed out. To continue working, log in.
+          </ModalBody>
+          <ModalFooter>
+            <Button
+              key="confirm"
+              variant="primary"
+              onClick={() => logout().then(() => window.location.reload())}
+            >
+              Log in
+            </Button>
+          </ModalFooter>
+        </Modal>
```

Comment:
Can you move this Modal to a separate file for better separation.

Reply:
Yes, thanks, created a separate component for this.